harness/determined/integrations/huggingface/hf_callback.py

```diff
@@ -27,26 +27,39 @@ def __init__(
         self.user_data = user_data
         self.load_last_checkpoint(args)
 
-        self.last_metrics: typing.Dict[str, float] = {"train_step": -1, "eval_step": -1}
-
-        searcher_config = det.get_cluster_info().trial._config["searcher"]
-        self.searcher_metric = searcher_config["metric"]
-        self.searcher_unit = list(searcher_config["max_length"].keys())[0]
-        self.searcher_max_length = list(searcher_config["max_length"].values())[0]
+        self.last_metrics: Dict[str, float] = {"train_step": -1, "eval_step": -1}
         self.searcher_ops = self.core_context.searcher.operations()
         self.current_op = next(self.searcher_ops)
-        self._check_searcher_compatibility(args)
         self.updating_searcher = False
 
+        cluster_info = det.get_cluster_info()
+        assert (
+            cluster_info
+        ), "Could not find `cluster_info`! The HF Callback must be run on a Determined Cluster!"
```

Comment:
don't use asserts to validate user behavior, because they can be disabled with some optimization flags to python (which is dumb, but that's how it is).

Asserts are still suitable for making mypy happy or validating internal assumptions.

Reply:
also let's drop the exclamation points from the error message, it's not really that exciting of a situation...

Reply:
Right, These are primarily to make mypy happy